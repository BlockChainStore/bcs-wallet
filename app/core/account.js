// @flow
import { wallet } from 'neon-js'

import { getStorage, setStorage } from './storage'

function getNEP6AddressData (matchesEncryptedWIF: boolean, address: string) {
  if (matchesEncryptedWIF) {
    return { address }
  } else {
    return {}
  }
}

export async function upgradeNEP6AddAddresses (encryptedWIF: string, wif: string) {
  const data = getStorage('userWallet')
  const loggedIntoAccount = new wallet.Account(wif)

  if (data && data.accounts) {
    const accounts = data.accounts.map((account, idx) => ({
      ...account,
      ...getNEP6AddressData(account.key === encryptedWIF, loggedIntoAccount.address)
    }))

    await setStorage('userWallet', { ...data, accounts })
  }
}
