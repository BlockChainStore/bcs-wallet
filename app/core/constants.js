// @flow
export const NEON_WALLET_RELEASE_LINK =
  'https://github.com/CityOfZion/neon-wallet/releases'

export const ASSETS = {
  NEO: 'NEO',
  GAS: 'GAS'
}

export const EXPLORERS = {
  NEO_TRACKER: 'Neotracker',
  NEO_SCAN: 'Neoscan',
  ANT_CHAIN: 'Antchain',
  NEO_VERSE: 'NEOverse'
}

export const ROUTES = {
  HOME: '',
  DASHBOARD: '/dashboard',
  CREATE_WALLET: '/create',
  ENCRYPT_KEY: '/encrypt-key',
  LOGIN_PRIVATE_KEY: '/login-private-key',
  LOGIN_LOCAL_STORAGE: '/login-local-storage',
  LOGIN_LEDGER_NANO_S: '/login-ledger-nano-s',
  LOGIN_NEP2: '/login-encrypted',
  LOGIN_TOKEN_SALE: '/login-token-sale',
  TOKEN_SALE: '/token-sale',
  SETTINGS: '/settings',
  DISPLAY_WALLET_KEYS: '/display-wallet-keys'
}

export const NOTIFICATION_LEVELS = {
  ERROR: 'error',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning'
}

export const NOTIFICATION_POSITIONS = {
  TOP_CENTER: 'tc',
  TOP_RIGHT: 'tr',
  TOP_LEFT: 'tl',
  BOTTOM_CENTER: 'bc',
  BOTTOM_RIGHT: 'br',
  BOTTOM_LEFT: 'bl'
}

export const BIP44_PATH =
  '8000002C' + '80000378' + '80000000' + '00000000' + '00000000'

export const MODAL_TYPES = {
  SEND: 'SEND',
  RECEIVE: 'RECEIVE',
  CONFIRM: 'CONFIRM',
  TOKEN_INFO: 'TOKEN_INFO',
  TOKEN: 'TOKEN',
  ICO: 'ICO'
}

export const MAIN_NETWORK_ID = '1'
export const TEST_NETWORK_ID = '2'

// TestNet
export const TOKENS_TEST = {
  BCS : 'e81c837d37763afbf894d319c868a09f219f2be9'
}

// MainNet
export const TOKENS = {
  BCS : '546c5872a992b2754ef327154f4c119baabff65f'
}

export const DEFAULT_WALLET = {
  name: 'userWallet',
  version: '1.0',
  scrypt: {
    cost: 16384,
    blockSize: 8,
    parallel: 8,
    size: 64
  },
  accounts: [],
  extra: null
}

export const DEFAULT_CURRENCY_CODE = 'usd'

export const CURRENCIES = {
  aud: { symbol: '$' },
  brl: { symbol: 'R$' },
  cad: { symbol: '$' },
  chf: { symbol: 'Fr.' },
  clp: { symbol: '$' },
  cny: { symbol: '¥' },
  czk: { symbol: 'Kč' },
  dkk: { symbol: 'kr. ' },
  eur: { symbol: '€' },
  gbp: { symbol: '£' },
  hkd: { symbol: '$' },
  huf: { symbol: 'Ft ' },
  idr: { symbol: 'Rp ' },
  ils: { symbol: '₪' },
  inr: { symbol: '₹' },
  jpy: { symbol: '¥' },
  krw: { symbol: '₩' },
  mxn: { symbol: '$' },
  myr: { symbol: 'RM' },
  nok: { symbol: 'kr ' },
  nzd: { symbol: '$' },
  php: { symbol: '₱' },
  pkr: { symbol: '₨ ' },
  pln: { symbol: 'zł' },
  rub: { symbol: '₽' },
  sek: { symbol: 'kr ' },
  sgd: { symbol: 'S$' },
  thb: { symbol: '฿' },
  try: { symbol: '₺' },
  twd: { symbol: 'NT$' },
  usd: { symbol: '$' },
  zar: { symbol: 'R ' }
}
