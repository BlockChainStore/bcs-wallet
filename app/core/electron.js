// @flow
import { shell } from 'electron'

export const openExternal = (srcLink: string) => {
  shell.openExternal(srcLink)
}
