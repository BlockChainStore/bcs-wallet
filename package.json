{
    "name": "BCSwallet",
    "version": "0.1.1",
    "main": "./main.js",
    "description": "BCSwallet folked from neon-wallet, Light wallet for NEO blockchain",
    "homepage": "https://github.com/CityOfZion/neon-wallet",
    "author": "Ethan Fast <ejhfast@gmail.com> (https://github.com/Ejhfast)",
    "scripts": {
        "postinstall": "electron-rebuild --force",
        "start": "cross-env NODE_ENV=production electron .",
        "assets": "cross-env NODE_ENV=production webpack --config ./config/webpack.config.prod",
        "dev": "cross-env START_HOT=1 NODE_ENV=development node_modules/.bin/webpack-dev-server --config ./config/webpack.config.dev",
        "start-dev": "cross-env NODE_ENV=development electron . --enable-logging --remote-debugging-port=9222",
        "assets-dev": "cross-env NODE_ENV=development webpack --config ./config/webpack.config.dev",
        "test": "npm rebuild node-hid && jest --coverage",
        "test-ci": "yarn test && cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js",
        "test:e2e": "ava __e2e__/*.e2e.js --timeout=120s",
        "test-watch": "jest --watch",
        "pack": "build --dir",
        "dist": "yarn assets && yarn build",
        "lint": "eslint -c .eslintrc ./app --ext .js,.jsx --ignore-pattern dist/",
        "lint-fix": "eslint -c .eslintrc ./app --ext .js,.jsx --fix --ignore-pattern dist/",
        "prepush": "echo 'running flow...' && yarn flow && echo 'running lint...' && yarn lint && echo 'running tests...' && yarn test",
        "electron-rebuild": "electron-rebuild --force --module_dir . -w node-hid",
        "flow": "flow check",
        "release-master": "./scripts/release-master.sh",
        "clean": "rm -r app/dist/*"
    },
    "keywords": [],
    "license": "ISC",
    "build": {
        "appId": "com.electron.neon",
        "files": [
            "./main.js",
            "app/dist/index.html",
            "app/dist/bundle.js",
            "app/dist/style.css"
        ],
        "dmg": {
            "contents": [{
                    "x": 110,
                    "y": 150
                },
                {
                    "x": 240,
                    "y": 150,
                    "type": "link",
                    "path": "/Applications"
                }
            ]
        },
        "linux": {
            "target": [
                "AppImage",
                "deb"
            ]
        },
        "win": {
            "target": "nsis",
            "icon": "build/icon.ico"
        }
    },
    "dependencies": {
        "axios": "0.17.0",
        "babel": "^6.23.0",
        "base-x": "3.0.2",
        "bigi": "1.4.2",
        "bignumber.js": "5.0.0",
        "bs58check": "2.0.2",
        "buffer": "5.0.8",
        "classnames": "2.2.5",
        "cleave.js": "1.0.7",
        "coveralls": "3.0.0",
        "crypto-js": "3.1.9-1",
        "ecurve": "1.0.5",
        "electron-context-menu": "0.9.1",
        "electron-json-storage": "4.0.2",
        "electron-save-file": "1.0.2",
        "elliptic": "6.4.0",
        "es6-promisify": "^5.0.0",
        "file-loader": "1.1.5",
        "fs": "0.0.1-security",
        "global": "4.3.2",
        "history": "4.7.2",
        "isomorphic-fetch": "2.2.1",
        "js-scrypt": "0.2.0",
        "lodash": "4.17.4",
        "neon-js": "git+https://github.com/cityofzion/neon-js.git#2.3.1",
        "node-gyp": "^3.6.2",
        "node-hid": "file:node-hid",
        "prop-types": "^15.6.0",
        "qrcode": "0.9.0",
        "raf": "3.4.0",
        "react": "16.1.1",
        "react-click-outside": "3.0.0",
        "react-data-grid": "2.0.73",
        "react-dom": "16.1.1",
        "react-hot-loader": "3.1.1",
        "react-icons": "2.2.7",
        "react-modal": "3.1.2",
        "react-notification-system": "0.2.16",
        "react-redux": "5.0.6",
        "react-router-dom": "4.2.2",
        "react-split-pane": "0.1.68",
        "react-syntax-highlighter": "5.8.0",
        "react-test-renderer": "16.1.1",
        "react-tippy": "1.2.2",
        "recompose": "^0.26.0",
        "redux": "3.7.2",
        "redux-logger": "3.0.6",
        "redux-saga": "^0.16.0",
        "redux-thunk": "2.2.0",
        "reselect": "^3.0.1",
        "secure-random": "1.1.1",
        "sifter": "0.5.3",
        "spawn-rx": "^2.0.12",
        "uuid": "^3.2.1",
        "wif": "2.0.6"
    },
    "devDependencies": {
        "autoprefixer": "7.1.6",
        "ava": "0.23.0",
        "axios-mock-adapter": "1.9.0",
        "babel-cli": "6.26.0",
        "babel-core": "6.26.0",
        "babel-eslint": "8.0.2",
        "babel-loader": "7.1.2",
        "babel-plugin-lodash": "3.2.11",
        "babel-plugin-transform-async-to-generator": "6.24.1",
        "babel-polyfill": "6.26.0",
        "babel-preset-env": "1.6.1",
        "babel-preset-flow": "6.23.0",
        "babel-preset-react": "6.24.1",
        "babel-preset-stage-0": "6.24.1",
        "css-loader": "0.28.7",
        "electron": "1.7.9",
        "electron-builder": "19.49.4",
        "electron-devtools-installer": "2.2.1",
        "electron-packager": "9.1.0",
        "electron-rebuild": "1.6.0",
        "enzyme": "3.1.1",
        "enzyme-adapter-react-16": "1.1.0",
        "enzyme-to-json": "3.2.1",
        "eslint": "^4.18.1",
        "eslint-config-standard": "10.2.1",
        "eslint-config-standard-react": "5.0.0",
        "eslint-loader": "1.9.0",
        "eslint-plugin-flowtype": "2.39.1",
        "eslint-plugin-import": "2.8.0",
        "eslint-plugin-jest": "21.2.0",
        "eslint-plugin-jsx-a11y": "6.0.2",
        "eslint-plugin-node": "5.2.1",
        "eslint-plugin-promise": "3.6.0",
        "eslint-plugin-react": "7.4.0",
        "eslint-plugin-standard": "3.0.1",
        "extract-text-webpack-plugin": "3.0.2",
        "flow-bin": "0.58.0",
        "html-webpack-plugin": "2.30.1",
        "husky": "0.14.3",
        "identity-obj-proxy": "3.0.0",
        "jest": "21.2.1",
        "json-loader": "0.5.7",
        "node-sass": "4.6.0",
        "redux-devtools-extension": "2.13.2",
        "redux-mock-store": "1.3.0",
        "resolve-url-loader": "2.2.0",
        "rimraf": "2.6.2",
        "sass-loader": "6.0.6",
        "spectron": "3.7.2",
        "style-loader": "0.19.0",
        "uglifyjs-webpack-plugin": "1.0.1",
        "url-loader": "0.6.2",
        "webpack": "3.8.1",
        "webpack-dev-server": "2.9.4"
    },
    "jest": {
        "moduleNameMapper": {
            "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
            "\\.(css|less|sass|scss)$": "identity-obj-proxy"
        },
        "testPathIgnorePatterns": [
            "__tests__/testHelpers.js",
            "__tests__/setupTests.js"
        ],
        "unmockedModulePathPatterns": [
            "node_modules/react/",
            "node_modules/enzyme/"
        ],
        "snapshotSerializers": [
            "enzyme-to-json/serializer"
        ],
        "setupTestFrameworkScriptFile": "<rootDir>/__tests__/setupTests.js"
    }
}