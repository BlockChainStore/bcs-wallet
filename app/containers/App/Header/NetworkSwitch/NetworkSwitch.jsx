// @flow
import React, { Component } from 'react'

import { getNetworks } from '../../../../core/networks'
import styles from '../Header.scss'

type Props = {
  networkId: string,
  onChange: Function,
  networks: Array<NetworkItemType>
}

export default class NetworkSwitch extends Component<Props> {
  static defaultProps = {
    networks: getNetworks()
  }

  render () {
    const { networkId, networks } = this.props
    return (
      <div id='network' className={styles.navBarItem}>
        <span className={styles.navBarItemLabel}>Running on</span>
        <select defaultValue={networkId} onChange={this.handleChange} className='networkSelector'>
          {networks.map(({ label, id }: NetworkItemType) =>
            <option key={`networkOption${id}`} value={id}>{label}</option>
          )}
        </select>
      </div>
    )
  }

  handleChange = (event: Object) => {
    this.props.onChange(event.target.value)
  }
}
