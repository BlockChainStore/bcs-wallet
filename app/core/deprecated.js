// @flow
import { get, omit, pick } from 'lodash'

import { ID as AUTH_ID } from '../actions/authActions'
import { ID as BALANCES_ID } from '../actions/balancesActions'
import { ID as NETWORK_ID } from '../actions/networkActions'
import { ASSETS } from '../core/constants'
import { findNetwork } from '../core/networks'

const PREFIX = 'api'

// TODO: Module functions that inspect state such as `doClaimNotify` should be refactored into
//       higher-order components that expose function props for performing these actions.  Data that
//       is normally provided by functions like `getNetwork` can instead be retrieved via the
//       `withData` higher-order component.

export const getNetworkId = (state: Object) => {
  return get(state, `${PREFIX}.${NETWORK_ID}.data`)
}

export const getNetwork = (state: Object) => {
  return getNetworkById(getNetworkId(state))
}

export const getNetworkById = (networkId: string) => {
  return findNetwork(networkId).network
}

export const getAddress = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.address`)
}

export const getWIF = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.wif`)
}

export const getSigningFunction = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.signingFunction`)
}

export const getPublicKey = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.publicKey`)
}

export const getIsHardwareLogin = (state: Object) => {
  return get(state, `${PREFIX}.${AUTH_ID}.data.isHardwareLogin`)
}

export const getNEO = (state: Object): string => {
  return getBalances(state)[ASSETS.NEO]
}

export const getGAS = (state: Object): string => {
  return getBalances(state)[ASSETS.GAS]
}

export const getTokenBalances = (state: Object): Object => {
  return omit(getBalances(state), ASSETS.NEO, ASSETS.GAS)
}

export const getAssetBalances = (state: Object): Object => {
  return pick(getBalances(state), ASSETS.NEO, ASSETS.GAS)
}

export const getBalances = (state: Object) => {
  return get(state, `${PREFIX}.${BALANCES_ID}.data`)
}
