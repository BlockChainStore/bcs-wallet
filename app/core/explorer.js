// @flow
import { openExternal } from './electron'
import { EXPLORERS } from '../core/constants'
import { isMainNetwork } from '../core/networks'

export const getExplorerBaseURL = (networkId: string, explorer: ExplorerType) => {
  let baseURL
  if (explorer === EXPLORERS.NEO_TRACKER) {
    if (isMainNetwork(networkId)) {
      baseURL = 'https://neotracker.io'
    } else {
      baseURL = 'https://testnet.neotracker.io'
    }
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    if (isMainNetwork(networkId)) {
      baseURL = 'https://neoscan.io'
    } else {
      baseURL = 'https://neoscan-testnet.io'
    }
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    if (isMainNetwork(networkId)) {
      baseURL = 'http://explorer.neoverse.io'
    } else {
      baseURL = 'http://testnet.neoverse.io'
    }
  } else {
    if (isMainNetwork(networkId)) {
      baseURL = 'http://antcha.in'
    } else {
      baseURL = 'http://testnet.antcha.in'
    }
  }
  return baseURL
}

export const getExplorerTxLink = (networkId: string, explorer: ExplorerType, txId: string) => {
  const baseURL = getExplorerBaseURL(networkId, explorer)

  if (explorer === EXPLORERS.NEO_TRACKER) {
    return `${baseURL}/tx/${txId}`
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    return `${baseURL}/transaction/${txId}`
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    return `${baseURL}/transactions/${txId}`
  } else {
    return `${baseURL}/tx/hash/${txId}`
  }
}

export const getExplorerAddressLink = (networkId: string, explorer: ExplorerType, address: string) => {
  const baseURL = getExplorerBaseURL(networkId, explorer)

  if (explorer === EXPLORERS.NEO_TRACKER) {
    return `${baseURL}/address/${address}`
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    return `${baseURL}/address/${address}/1`
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    return `${baseURL}/addresses/${address}`
  } else {
    return `${baseURL}/address/info/${address}`
  }
}

export const getExplorerAssetLink = (networkId: string, explorer: ExplorerType, assetId: string) => {
  const baseURL = getExplorerBaseURL(networkId, explorer)

  if (explorer === EXPLORERS.NEO_TRACKER) {
    return `${baseURL}/asset/${assetId}`
  } else if (explorer === EXPLORERS.NEO_SCAN) {
    return `${baseURL}/asset/${assetId}`
  } else if (explorer === EXPLORERS.NEO_VERSE) {
    return `${baseURL}/assets/${assetId}`
  } else {
    return `${baseURL}/asset/hash/${assetId}`
  }
}

export const openExplorerTx = (networkId: string, explorer: ExplorerType, txId: string) =>
  openExternal(getExplorerTxLink(networkId, explorer, txId))

export const openExplorerAddress = (networkId: string, explorer: ExplorerType, address: string) =>
  openExternal(getExplorerAddressLink(networkId, explorer, address))

export const openExplorerAsset = (networkId: string, explorer: ExplorerType, assetId: string) =>
  openExternal(getExplorerAssetLink(networkId, explorer, assetId))
