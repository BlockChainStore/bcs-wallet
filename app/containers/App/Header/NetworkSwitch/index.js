// @flow
import { compose } from 'recompose'

import NetworkSwitch from './NetworkSwitch'
import networkActions from '../../../../actions/networkActions'
import withActions from '../../../../hocs/api/withActions'
import withNetworkData from '../../../../hocs/withNetworkData'
import { type Actions } from '../../../../values/api'

const mapActionsToProps = (actions: Actions, props: Object): Object => ({
  onChange: (networkId) => actions.request({ networkId })
})

export default compose(
  withNetworkData(),
  withActions(networkActions, mapActionsToProps)
)(NetworkSwitch)
