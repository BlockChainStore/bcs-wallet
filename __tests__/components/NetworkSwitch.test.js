import React from 'react'
import { shallow } from 'enzyme'

import NetworkSwitch from '../../app/containers/App/Header/NetworkSwitch/NetworkSwitch'
import { MAIN_NETWORK_ID, TEST_NETWORK_ID } from '../../app/core/constants'

// TODO research how to move the axios mock code which is repeated in NetworkSwitch to a helper or config file
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { version } from '../../package.json'

const axiosMock = new MockAdapter(axios)
axiosMock
  .onGet('http://testnet-api.wallet.cityofzion.io/v2/version')
  .reply(200, { version })
axiosMock
  .onGet('https://api.coinmarketcap.com/v1/ticker/NEO/', { params: { convert: 'USD' } })
  .reply(200, [ { price_usd: 24.50 } ])
axiosMock
  .onGet('https://api.coinmarketcap.com/v1/ticker/GAS/', { params: { convert: 'USD' } })
  .reply(200, [ { price_usd: 18.20 } ])
jest.mock('neon-js')

const setup = () => {
  const props = {
    networkId: MAIN_NETWORK_ID,
    networks: [
      {
        id: MAIN_NETWORK_ID,
        label: 'MainNet',
        network: 'MainNet'
      },
      {
        id: TEST_NETWORK_ID,
        label: 'TestNet',
        network: 'TestNet'
      }
    ],
    onChange: jest.fn()
  }
  const wrapper = shallow(<NetworkSwitch {...props} />)

  return {
    wrapper
  }
}

describe('NetworkSwitch', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('correctly renders MainNet initially', () => {
    const { wrapper } = setup()
    const networkSelectorElement = wrapper.find('.networkSelector').getElement()

    expect(networkSelectorElement.props.defaultValue).toEqual(MAIN_NETWORK_ID)
  })

  test('switches to the correct network when chosen from the dropdown', () => {
    const { wrapper } = setup()
    const instance = wrapper.instance()
    const networkSelector = wrapper.find('.networkSelector')

    networkSelector.simulate('change', { target: { value: TEST_NETWORK_ID } })
    expect(instance.props.onChange).toHaveBeenCalledWith(TEST_NETWORK_ID)

    networkSelector.simulate('change', { target: { value: MAIN_NETWORK_ID } })
    expect(instance.props.onChange).toHaveBeenCalledWith(MAIN_NETWORK_ID)
  })
})
