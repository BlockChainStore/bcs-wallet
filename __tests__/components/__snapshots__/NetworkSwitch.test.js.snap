// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkSwitch renders without crashing 1`] = `
<div
  className="navBarItem"
  id="network"
>
  <span
    className="navBarItemLabel"
  >
    Running on
  </span>
  <select
    className="networkSelector"
    defaultValue="1"
    onChange={[Function]}
  >
    <option
      key="networkOption1"
      value="1"
    >
      MainNet
    </option>
    <option
      key="networkOption2"
      value="2"
    >
      TestNet
    </option>
  </select>
</div>
`;
